components/script/dom/servoparser/html.rs

```diff
@@ -210,6 +210,12 @@ impl<'a> Serialize for &'a Node {
     ) -> io::Result<()> {
         let node = *self;
 
+        if let TraversalScope::ChildrenOnly(_) = traversal_scope {
+            if node.downcast::<Element>().unwrap().is_void() {
```

Comment:
We should use `if node.downcast::<Element>().map_or(false, |e| e.is_void()) {` instead, so we do not panic when serializing non-element nodes.